Data/Matrix.hs

```diff
@@ -556,6 +556,31 @@ inverse m
             rref'd = rref adjoinedWId
         in rref'd >>= return . submatrix 1 (nrows m) (ncols m + 1) (ncols m * 2)
 
+-- | /O(strassenMixed)/. The inverse of a square matrix.
+--   Inverts blockwise. Significantly faster than naive Gaussian elimination.
+--   Same time complexity as whatever matrix multiplication (*) currently uses.
+--   See https://en.wikipedia.org/wiki/Invertible_matrix#Blockwise_inversion
+blockInverse :: (Fractional a, Eq a) => Matrix a -> Either String (Matrix a)
+blockInverse m
+    | nrows m /= ncols m = inverse m
+    | otherwise = Right $ blockInverseUnsafe m
+
+-- blockInverse that assumes that it is called
+-- on a square matrix.
+blockInverseUnsafe :: (Fractional a, Eq a) => Matrix a -> Matrix a
+blockInverseUnsafe m
+    | nrows m < 10 = either (\_ -> m) id $ inverse m
```

Comment:
Sorry to waste your time – I've realized I made a boneheaded error. I used this code to speed up a problem I was having with positive definite matrices – but of course not all matrices are positive definite. However, since almost all matrices are invertible, I think we could still get a significant improvement for most matrices by using this algorithm then falling back if it gets stuck. Alternatively, I may try to implement something like in this paper (http://www.sciencedirect.com/science/article/pii/S0377042708006237) but I'm not sure whether it'll actually yield that much of a speedup.

I noticed you're not super concerned about the speed of this library, and a consistent improvement on the inversion algorithm would deal with a lot of special-casing and tradeoffs. If you don't think this would be a major improvement you can go ahead and close the PR. 

Anyway, thanks for the library! Super useful.